packages/api-bindings/src/apollo/links.ts

```diff
@@ -0,0 +1,107 @@
+// eslint-disable-next-line no-restricted-imports
+import { from, fromPromise, HttpLink } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context';
+import { onError } from '@apollo/client/link/error';
+import { maybe } from '@apollo/client/utilities';
+import { ILogger, never } from '@lens-protocol/shared-kernel';
+
+import { SemVer } from '../SemVer';
+import { IAccessTokenStorage } from './IAccessTokenStorage';
+
+/**
+ * An error code that's coming from `apollo-server-errors` `AuthenticationError`
+ */
+const AUTHENTICATION_ERROR_CODE = 'UNAUTHENTICATED';
+
+export function createAuthLink(accessTokenStorage: IAccessTokenStorage) {
+  const errorLink = onError(({ graphQLErrors, operation, forward }) => {
+    if (
+      graphQLErrors &&
+      graphQLErrors.some((error) => error.extensions?.code === AUTHENTICATION_ERROR_CODE)
+    ) {
+      return fromPromise(accessTokenStorage.refreshToken()).flatMap(() => forward(operation));
+    }
+    return;
+  });
+
+  const authHeaderLink = setContext(() => {
+    const token = accessTokenStorage.getAccessToken();
+
+    if (token) {
+      return {
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      };
+    }
+
+    return;
+  });
+
+  return from([errorLink, authHeaderLink]);
+}
+
+const backupFetch = maybe(() => fetch);
+
+function wrapFetch(
+  logger: ILogger,
+  supportedVersion: SemVer,
+  fetch: WindowOrWorkerGlobalScope['fetch'],
+): WindowOrWorkerGlobalScope['fetch'] {
+  return async (...args) => {
+    const response = await fetch(...args);
+
+    if (response.status === 200) {
+      const apiVersion = response.headers.get('x-api-version');
+
+      if (apiVersion) {
+        if (apiVersion < supportedVersion) {
+          logger.error(
+            `The Lens API ${apiVersion} is outside of the Lens SDK support range ^${supportedVersion}`,
+          );
+          return response;
+        }
+
+        const [apiMajor, apiMinor] = apiVersion.split('.');
+        const [clientMajor, clientMinor] = supportedVersion.split('.');
+
+        if (apiMajor && clientMajor && apiMajor > clientMajor) {
+          logger.warn(
+            `The Lens API ${apiVersion} is NOT supported by the Lens SDK support range ^${supportedVersion}. Update your Lens SDK client to the latest version.`,
+          );
+          return response;
+        }
+
+        if (apiMinor && clientMinor && apiMinor > clientMinor) {
+          logger.info(
+            `The Lens API ${apiVersion} is ahead of the Lens SDK support range ^${supportedVersion}. Check for a new version of the Lens SDK client, if available.`,
```

Comment:
so it will already log as the current API version is `x-api-version: 1.31.0` and we start here with `1.0.0`, yes?